packages/descendants/src/DescendantsContext.tsx

```diff
@@ -0,0 +1,33 @@
+import { createContext, Dispatch } from 'react';
+
+import { DescendantsList } from './Descendants.types';
+import { DescendantsReducerAction } from './DescendantsReducer';
+
+export type RegisterDescendantFn<T extends HTMLElement> = (
+  element?: T | null,
+) => () => void;
+
+export interface DescendantsContextProps<T extends HTMLElement> {
+  descendants: DescendantsList<T>;
+  dispatch: Dispatch<DescendantsReducerAction<T>>;
+}
+
+export type DescendantContextType<T extends HTMLElement> = React.Context<
```

Comment:
do these interfaces belong in .types.ts

Reply:
Potentially... I was imagining `.types.ts` would have shared types, and files like this would have types specific to the use case. 
(If these were more complex files, they should each have their own folder with a `Context.types.ts`, `Context.spec.ts` etc.)